bindings/node.js/kzg.cxx

```diff
@@ -118,62 +153,26 @@ inline Bytes48 *get_bytes48(const Napi::Env &env, const Napi::Value &val, std::s
   return reinterpret_cast<Bytes48 *>(get_bytes(env, val, BYTES_PER_COMMITMENT, name));
 }
 
-// loadTrustedSetup: (filePath: string) => SetupHandle;
 Napi::Value LoadTrustedSetup(const Napi::CallbackInfo& info) {
-  auto env = info.Env();
-
-  size_t argument_count = info.Length();
-  size_t expected_argument_count = 1;
-  if (argument_count != expected_argument_count) {
-    return throw_invalid_arguments_count(expected_argument_count, argument_count, env);
-  }
-
-  if (!info[0].IsString()) {
-    return throw_invalid_argument_type(env, "filePath", "string");
-  }
-
-  const std::string file_path = info[0].ToString().Utf8Value();
-
-  KZGSettings* kzg_settings = (KZGSettings*)malloc(sizeof(KZGSettings));
-
-  if (kzg_settings == NULL) {
-    Napi::Error::New(env, "Error while allocating memory for KZG settings").ThrowAsJavaScriptException();
-    return env.Null();
-  }
-
-  FILE* f = fopen(file_path.c_str(), "r");
-
-  if (f == NULL) {
-    free(kzg_settings);
-    Napi::Error::New(env, "Error opening trusted setup file: " + file_path).ThrowAsJavaScriptException();
-    return env.Null();
-  }
-
-  if (load_trusted_setup_file(kzg_settings, f) != C_KZG_OK) {
-    free(kzg_settings);
-    Napi::Error::New(env, "Error loading trusted setup file").ThrowAsJavaScriptException();
-    return env.Null();
-  }
-
-  return Napi::External<KZGSettings>::New(info.Env(), kzg_settings);
-}
-
-// freeTrustedSetup: (setupHandle: SetupHandle) => void;
-Napi::Value FreeTrustedSetup(const Napi::CallbackInfo& info) {
-  auto env = info.Env();
-
-  size_t argument_count = info.Length();
-  size_t expected_argument_count = 1;
-  if (argument_count != expected_argument_count) {
-    return throw_invalid_arguments_count(expected_argument_count, argument_count, env);
-  }
-  KZGSettings *kzg_settings = get_kzg_settings(env, info[0]);
-  if (kzg_settings == nullptr) {
-    return env.Null();
-  }
-
-  free_trusted_setup(kzg_settings);
-  free(kzg_settings);
+  Napi::Env env = info.Env();
+  KzgBindings *bindings = static_cast<KzgBindings *>(info.Data());
+  if (bindings->IsSetup()) {
+      // QUESTION: Should this throw for re-setup or just ignore like it is?
```

Comment:
Seems like something I would throw an exception for, but no strong opinion.

Reply:
Yeah I would throw an exception here.

Reply:
Updated.